packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,47 +5,48 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {
-  useState,
-  useEffect,
-  useContext,
-  useRef,
-  useCallback,
-} from 'react';
+import React, {useState, useRef, useCallback} from 'react';
 import classnames from 'classnames';
 
-import DocusaurusContext from '@docusaurus/context';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 
 import './styles.css';
 
+const loadJS = () => import('docsearch.js');
+let loadedJs = false;
+
 const Search = props => {
-  const [isEnabled, setIsEnabled] = useState(true);
+  const [initialized, setInitialized] = useState(false);
   const searchBarRef = useRef(null);
-  const context = useContext(DocusaurusContext);
+  const {siteConfig = {}} = useDocusaurusContext();
+  const {
+    themeConfig: {algolia},
+  } = siteConfig;
 
-  useEffect(() => {
-    const {siteConfig = {}} = context;
-    const {
-      themeConfig: {algolia},
-    } = siteConfig;
+  const initAlgolia = () => {
+    if (!initialized) {
+      window.docsearch({
+        appId: algolia.appId,
+        apiKey: algolia.apiKey,
+        indexName: algolia.indexName,
+        inputSelector: '#search_input_react',
+        algoliaOptions: algolia.algoliaOptions,
+      });
+      setInitialized(true);
+    }
+  };
 
-    // https://github.com/algolia/docsearch/issues/352
-    const isClient = typeof window !== 'undefined';
-    if (isClient) {
-      import('docsearch.js').then(({default: docsearch}) => {
-        docsearch({
-          appId: algolia.appId,
-          apiKey: algolia.apiKey,
-          indexName: algolia.indexName,
-          inputSelector: '#search_input_react',
-          algoliaOptions: algolia.algoliaOptions,
-        });
+  const loadAlgoliaJS = () => {
+    if (!loadedJs) {
+      loadJS().then(a => {
```

Comment:
Wanna add some error handling?

Reply:
I dont know what kind of error should be introduced. Previously doesnt have it too.